src/plugins/intel_gna/src/common/gna_target.hpp

```diff
@@ -4,16 +4,48 @@
 
 #pragma once
 
+#include <string>
+
+#include "gna2-common-api.h"
+#include "openvino/runtime/intel_gna/properties.hpp"
+
 namespace ov {
 namespace intel_gna {
 namespace common {
 
-static constexpr const char* kGnaTargetUnspecified = "";
-static constexpr const char* kGnaTarget2_0 = "GNA_TARGET_2_0";
-static constexpr const char* kGnaTarget3_0 = "GNA_TARGET_3_0";
-static constexpr const char* kGnaTarget3_1 = "GNA_TARGET_3_1";
-static constexpr const char* kGnaTarget3_5 = "GNA_TARGET_3_5";
-static constexpr const char* kGnaDefaultTarget = kGnaTarget3_0;
+enum class DeviceVersion {
```

Comment:
I would initialize this values with values from Gna2DeviceVersion or event better create conversion function/operator
from Gna2DeviceVersion to DeviceVersion  and vice versa.

further you are sing casting from DeviceVersion  to Gna2DeviceVersion .
and e.g.

DeviceVersion::NotSet is -1 which does not map to any value Gna2DeviceVersion - it may lead to undefined behaviour